tests/functional/test_list.py

```diff
@@ -11,18 +16,97 @@ def test_list_command(script, data):
         'install', '-f', data.find_links, '--no-index', 'simple==1.0',
         'simple2==3.0',
     )
-    result = script.pip('list')
+    result = script.pip('list', expect_stderr=True)
+    assert WARN_NOCOL in result.stderr, str(result)
```

Comment:
Hmm, this may explain all the above - are you getting the "```--no-columns``` will go away" warning for the default usage? I don't think you should - I think it should only happen if the user *explicitly* requests ```--no-columns```.